File: src/main/java/itmo/is/project/model/module/BuildCost.java
package itmo.is.project.model.module;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "build_cost")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BuildCost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @OneToMany(mappedBy = "buildCost", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private List<BuildCostItem> items;
}

File: src/main/kotlin/itmo/isproject/model/module/BuildCost.kt
package itmo.isproject.model.module

import itmo.isproject.model.IntIdEntity
import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "build_cost")
class BuildCost(

    @OneToMany(mappedBy = "buildCost", fetch = FetchType.EAGER, cascade = [CascadeType.ALL])
    var items: MutableList<BuildCostItem> = ArrayList()

) : IntIdEntity()
